pkg/sql/distsqlrun/indexbackfiller.go

```diff
@@ -135,13 +136,36 @@ func (ib *indexBackfiller) runChunk(
 		sortTime := timeutil.Now().Sub(start)
 
 		start = timeutil.Now()
-		adder, err := ib.flowCtx.BulkAdder(ctx, ib.flowCtx.ClientDB, 32<<20, readAsOf)
+		const maxFlushBytes = 32 << 20
+		flushBytes := int64(len(entries)) * 1000
+		if flushBytes > maxFlushBytes {
+			flushBytes = maxFlushBytes
+		}
+		adder, err := ib.flowCtx.BulkAdder(ctx, ib.flowCtx.ClientDB, flushBytes, readAsOf)
 		if err != nil {
 			return nil, err
 		}
 		defer adder.Close()
+		containsInvertedIndex := ib.ContainsInvertedIndex()
 		for i := range entries {
 			if err := adder.Add(ctx, entries[i].Key, entries[i].Value.RawBytes); err != nil {
+				// Detect a duplicate within the SST being constructed. This is an
+				// insufficient method for unique constraint enforcement and the index
+				// has to be validated after construction.
+				if strings.Contains(err.Error(), "Keys must be added in order") {
```

Comment:
I don't know if it matters, but what I did for this in my IMPORT prototype is check `i > 0 && bytes.Equal(entries[i-1].Key, entries[i].Key)` instead of inspecting the error string

Reply:
done